lib/fs.js

```diff
@@ -60,9 +60,21 @@ module.exports = {
 
     let fileSystem;
     // store our files in memory
-    const isMemoryFs = !compiler.compilers && compiler.outputFileSystem instanceof MemoryFileSystem;
+    const isConfiguredFs = context.options.fs;
+    const isMemoryFs = !isConfiguredFs
+      && !compiler.compilers
+      && compiler.outputFileSystem instanceof MemoryFileSystem;
 
-    if (isMemoryFs) {
+    if (isConfiguredFs) {
+      const { fs } = context.options;
+      if (typeof fs.join !== 'function') {
```

Comment:
Just questions, can you find place where we use `fs.join`? Looks like we should remove this/deprecated from code base.

Reply:
It is not used by the `webpack-middleware` but rather by `wepback` instead. Here it is as a precaution to avoid digging up webpack's erros. If you attempt to provide `fs` without `join()` webpack will complain about it when accessing `webpack.outputFileSystem`.

I can switch to Russian to explain more if you like.

Reply:
> I can switch to Russian to explain more if you like.

No need

> It is not used by the `webpack-middleware` but rather by `wepback` instead. Here it is as a precaution to avoid digging up webpack's erros. If you attempt to provide `fs` without `join()` webpack will complain about it when accessing `webpack.outputFileSystem`.

I think we should open issue in webpack and remove all `fs.join` in `webpack@5`, `join` is not part of `fs` and looks very dirty and misleading

Reply:
Well I got no time for that. May be next time.

Reply:
@hinell i mean just create issue in webpack repo and all, i am from mobile and it is hard for me right now